snuba/utils/streams/backends/kafka.py

```diff
@@ -65,42 +66,10 @@ def __init__(self, state: KafkaConsumerState):
 Headers = Sequence[Tuple[str, bytes]]
 
 
-class KafkaPayload:
-    # XXX: This is not a dataclass since dataclasses do not support classes
-    # with __slots__ *and* default values. Since we create a lot of these
-    # objects, it's probably more important to preserve their performance and
-    # memory impact than it is their developer friendliness (unfortunately.)
-
-    __slots__ = ["__key", "__value", "__headers"]
-
-    def __init__(
-        self, key: Optional[bytes], value: bytes, headers: Optional[Headers] = None
-    ) -> None:
-        self.__key = key
-        self.__value = value
-        self.__headers = headers if headers is not None else []
```

Comment:
`headers` now must be passed explicitly, as the old `__new__` trick no longer works with `NamedTuple`: https://github.com/python/typing/issues/526

In an ideal scenario, `KafkaPayload` would only be instantiated by the stream backend and codec instances so this should become less and less of a problem over time.